### PR TITLE
feat(client): isSet, updateMany, deleteMany, and filter re-write

### PR DESCRIPTION
**Pull request by millsp**
Integrates https://github.com/prisma/prisma-engines/pull/2797

### CODE REVIEW

File: packages/client/src/__tests__/integration/happy/composites-mongo/update/list.ts [hunk 1]
```diff
@@ -242,12 +242,108 @@ describeIf(!process.env.TEST_SKIP_MONGODB)('update > list', () => {
...
+      }
+    `)
+  })
 
   /**
    * Simple deleteMany
    */
-  test.skip('deleteMany', async () => {})
+  test('deleteMany', async () => {
+    const comment = await prisma.commentRequiredList.update({
```
Comment (Jolg42) [inline packages/client/src/__tests__/integration/happy/composites-mongo/update/list.ts:305]: test says `deleteMany` but client does `update`

File: packages/client/src/__tests__/integration/happy/composites-mongo/update/list.ts [hunk 2]
```diff
@@ -242,12 +242,108 @@ describeIf(!process.env.TEST_SKIP_MONGODB)('update > list', () => {
...
+                userId: 10,
+                vote: true,
+              },
+            ],
+          },
+          Object {
+            text: Hello World,
+            upvotes: Array [
+              Object {
+                userId: Another Comment,
```
Comment (Jolg42) [inline packages/client/src/__tests__/integration/happy/composites-mongo/update/list.ts:289]: Nit: it's a bit weird to mix a number and a string here for a userId.

I see the type is `userId String` in the schema

**Review by Jolg42 — APPROVED** [review]
Looks good! 

⚠️ don't merge as is because it's using an integration engine version

File: packages/client/src/__tests__/integration/happy/composites-mongo/update/list.ts (hunk 2 above)
Comment (millsp) [reply packages/client/src/__tests__/integration/happy/composites-mongo/update/list.ts:289]: Nice catch, thanks! These will be caught when we have automated typechecks (coming soon).

**Pull request closed without merging**